tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/NLog/LogsInjection/DiagnosticContextHelper.cs

```diff
@@ -127,12 +115,46 @@ public static void RemoveMdcState(MappedDiagnosticsContextSetterProxy mdc, bool
             }
         }
 
+        private static IReadOnlyList<KeyValuePair<string, object>> CreateEntriesList(Tracer tracer, out bool removeTraceIds)
```

Comment:
Could the method become like this:
```c#
        private static IReadOnlyList<KeyValuePair<string, object>> CreateEntriesList(Tracer tracer, out bool removeTraceIds)
        {
            // For mismatch version support we need to keep requesting old keys.
            var spanContext = tracer.DistributedSpanContext;
            var hasTraceId = spanContext is not null && (spanContext.TryGetValue(SpanContext.Keys.TraceId, out string traceId) ||
                                 spanContext.TryGetValue(HttpHeaderNames.TraceId, out traceId));
            var hasSpanId = spanContext is not null && (spanContext.TryGetValue(SpanContext.Keys.ParentId, out string spanId) ||
                                spanContext.TryGetValue(HttpHeaderNames.ParentId, out spanId));
            removeTraceIds = hasTraceId || hasSpanId;

            var entries =  !removeTraceIds ? new [] {
                  new KeyValuePair<string, object>(CorrelationIdentifier.ServiceKey, tracer.DefaultServiceName ?? string.Empty),
                  new KeyValuePair<string, object>(CorrelationIdentifier.VersionKey, tracer.Settings.ServiceVersion ?? string.Empty),
                  new KeyValuePair<string, object>(CorrelationIdentifier.EnvKey, tracer.Settings.Environment ?? string.Empty),
            } :
            {
                  new KeyValuePair<string, object>(CorrelationIdentifier.ServiceKey, tracer.DefaultServiceName ?? string.Empty),
                  new KeyValuePair<string, object>(CorrelationIdentifier.VersionKey, tracer.Settings.ServiceVersion ?? string.Empty),
                  new KeyValuePair<string, object>(CorrelationIdentifier.EnvKey, tracer.Settings.Environment ?? string.Empty),
                  new KeyValuePair<string, object>(CorrelationIdentifier.TraceIdKey, traceId),
                  new KeyValuePair<string, object>(CorrelationIdentifier.SpanIdKey, spanId),
            };
            return entries;
        }
```
Would remove allocation of List-array. Would remove re-allocation of List-array (when more than 4 items). Would remove allocation of ReadOnlyCollection